Process data into expression and mutation matrices

dhimmel opened the pull request.

dhimmel committed:
Design data/notebook repo, Download TCGA data

First in a series of commits to address cognoma/cognoma#14 — providing access to
the cancer data for Project Cognoma.

This initial commit downloads the files referenced by @gwaygenomics in
https://github.com/cognoma/cognoma/issues/14#issuecomment-232388435

Create README, conda environment, and script to perform the entire analysis.

dhimmel committed:
Process TCGA data into sample by gene matrices

dhimmel committed:
Use bash for execute script and exit on error

dhimmel committed:
Change download URL & store metadata for Xena

Changed `base_url` for downloading data from the Xena browser from
https://genome-cancer.ucsc.edu/download/public/xena/TCGA/TCGA.PANCAN.sampleMap/
to https://tcga.xenahubs.net/download/TCGA.PANCAN.sampleMap/. This new location
seems to have resolved the unstandardized mutation effects reported in
cognoma/cancer-data#2.

Added json metadata files to `download` providing version info at time of
download. Thanks @jingchunzhu for the suggestion. See
https://groups.google.com/forum/#!msg/ucsc-cancer-genomics-browser/eg6nJOFSefw/wO0wNrMeAgAJ

dhimmel committed:
Retain all code red & blue mutations

Addresses cognoma/cancer-data#2 -- add additional mutation effects. Added all
red & blue mutations from http://xena.ucsc.edu/how-we-characterize-mutations/
that were present in the data.

cgreene commented on execute.sh:
```diff
@@ -0,0 +1,16 @@
+#!/bin/bash
```
reviewed - running conda environment

cgreene commented on environment.yml:
```diff
@@ -0,0 +1,66 @@
+name: cognoma-cancer-data
```
dependencies - looks reasonable

cgreene commented:
The 2.TCGA-process.ipynb notebook says "Notice that mutation type codes are not standardized. For example, 5_prime_UTR_variant, 5'UTR, UTR_5_PRIME all refer to the same type of mutation." but the corresponding table doesn't have these elements. Did this get cleaned up before the table is generated somehow?

clairemcleod commented:
It seems like there is a bug with the exported .yml file and environment creation. `execute.sh` does not actually create an environment. It seems related to [this conda issue](https://github.com/conda/conda-env/issues/247).

clairemcleod commented on environment.yml:
```diff
@@ -0,0 +1,66 @@
+name: cognoma-cancer-data
+dependencies:
```
Some of these package requirements are specific to conda on linux. When `execute.sh` runs on osx, yields error due to crossplatform issue:
Error: Packages missing in current osx-64 channels: 
- fontconfig 2.11.1 6
- libsodium 1.0.10 0
- zeromq 4.1.4 0

(edited for clarity)

dhimmel replied:
@clairemcleod interesting error. It looks like `zeromq 4.1.4` is available for Linux, which I used to make the PR, but not for OS X in the [anaconda channel](https://anaconda.org/anaconda/zeromq/files?page=1). 4.1.3 is available for OS X.

It looks like others are having [similar issues](http://stackoverflow.com/q/38400676/4651668) recently. Options are to downgrade the versions for OS X compatibility, or to [notify anaconda](https://github.com/ContinuumIO/anaconda-issues/issues) and see if they can fix this.

clairemcleod replied:
@dhimmel It seems like it is the same issue with anaconda channel availability for the other two as well:
[`libsodium`](https://anaconda.org/anaconda/libsodium/files) - libsodium 1.0.3 is the most recent version available on linux/windows/osx
[`fontconfig`](https://anaconda.org/anaconda/fontconfig/files) - only available for linux? not sure what to do about this

dhimmel replied:
I submitted an issue to anaconda: https://github.com/ContinuumIO/anaconda-issues/issues/925. Let's see how that goes.

dhimmel replied:
@clairemcleod, I attempted to fix the issue (see https://github.com/cognoma/cancer-data/pull/7/commits/8fb66e94fec1b1df48121f30b45c3a492082c0a3). Do you want to see whether the environment works for you on OS X now?

clairemcleod replied:
@dhimmel This seems to work now - thanks!

dhimmel commented:
> Did this get cleaned up before the table is generated somehow?

@cgreene see https://github.com/cognoma/cancer-data/issues/2#issuecomment-233088248 -- the input data changed, resolving the issue. Will update the notebook.

dhimmel committed:
Remove old note on unstandardized mutation types

Unstandardized mutation types were resolved. See
https://github.com/cognoma/cancer-data/issues/2#issuecomment-233088248

Addressed this reviewer comment:
https://github.com/cognoma/cancer-data/pull/7#issuecomment-233801064

dhimmel committed:
Specify only explicit dependencies in environment

Switch environment.yml from specifying every installed package to just listing
explicit dependencies (packages users directly import or interact with). Remove
build numbers from version specification.

Aims to address ContinuumIO/anaconda-issues#925 where an environment created on
Linux was not available on OS X.

dhimmel committed:
Export notebooks to scripts

dhimmel committed:
Update doc /filtering code for duplicate mutations

Thanks @jingchunzhu for the relevant documentation on the Xena Browser Google
Group. See https://groups.google.com/d/msg/ucsc-cancer-genomics-browser/eg6nJOFSefw/Z0BM6pU9BAAJ:

> regarding the duplicated rows, they are due to that there are multiple
sequencing results for these samples, from the  multiple TCGA aliquots map to a
single sample. We left the duplications in the file intentionally, if you see
the same mutation is called on two rows, it is essentially a biological
replicate.  And you should also see occasionally in these duplications, majority
of them has, say for example, two exact same rows for sample A, but sometime
there is only one row of mutation B for sample A, which means that mutation B is
not detected in the biological replicate.

dhimmel committed:
Execute notebooks inplace before exporting scripts

This way cell numbers in the script comments will be sequential (refreshed).

cgreene commented:
My comments have been addressed 👍

dhimmel commented:
Cool @clairemcleod, would you like to merge if you think it's ready?

You'll want to use a "merge commit" rather than a "squash and merge" since there are multiple distinct commits. This is the default, but just wanted to clarify.
